models/ResourceService.cfc

```diff
@@ -1,439 +1,517 @@
+	 * @rbLocale The locale of the resource bundle
+	 */
+	struct function getResourceBundle( required rbFile, rbLocale = "en_US" ) {
+		var resourceBundle = {};
+		var thisKEY        = "";
+		var rbFilePath     = arguments.rbFile & iIf(
+			len( arguments.rbLocale ),
+			de( "_" ),
+			de( "" )
+		) & arguments.rbLocale & ".properties";
+		var rbFullPath = rbFilePath;
+
+		// Try to locate the path using the coldbox plugin utility
+		rbFullPath = variables.controller.locateFilePath( rbFilePath );
+
+		// Validate Location
+		if ( !len( rbFullPath ) ) {
+			throw(
+				"The resource bundle file: #rbFilePath# does not exist. Please check your path",
+				"FullPath: #rbFullPath#",
+				"ResourceBundle.InvalidBundlePath"
+			);
+		}
+
+		// create a file input stream with file location
+		var fis = createObject( "java", "java.io.FileInputStream" ).init( rbFullPath );
+		var fir = createObject( "java", "java.io.InputStreamReader" ).init( fis, "UTF-8" );
+		// Init RB with file Stream
+		var rb  = createObject( "java", "java.util.PropertyResourceBundle" ).init( fir );
+		try {
+			// Get Keys
+			var keys = rb.getKeys();
+
+			// Loop through property keys and store the values into bundle
+			while ( keys.hasMoreElements() ) {
+				thisKEY                   = keys.nextElement();
+				resourceBundle[ thisKEY ] = rb.handleGetObject( thisKEY );
 			}
-			catch(Any e){
-				fis.close();
-				$rethrow( e );
+		} catch ( Any e ) {
+			fis.close();
+			$rethrow( e );
+		}
+
+		// Close the input stream
+		fis.close();
+
+		return resourceBundle;
+	}
+
+	/**
+	 * Returns a given key from a specific resource bundle file and locale. NOT FROM MEMORY
+	 *
+	 * @rbFile This must be the path + filename UP to but NOT including the locale. We auto-add the local and .properties to the end.
+	 * @rbKey The key to retrieve
+	 * @rbLocale The locale of the bundle. Default is en_US
+	 * @defaultValue A default value to send back if resource not found
+	 *
+	 * @throws ResourceBundle.InvalidBundlePath if bundlePath is not found
+	 * @throws ResourceBundle.RBKeyNotFoundException if rbKey is not found
+	 */
+	any function getRBString(
+		required rbFile,
+		required rbKey,
+		rbLocale = "en_US",
+		defaultValue
+	) {
+		// default locale?
+		if ( !len( arguments.rbLocale ) ) {
+			arguments.rbLocale = variables.defaultLocale;
+		}
+
+		// prepare the file
+		var rbFilePath = arguments.rbFile & "_#arguments.rbLocale#.properties";
+
+		// Try to locate the path using the coldbox plugin utility
+		var rbFullPath = variables.controller.locateFilePath( rbFilePath );
+
+		// Validate Location
+		if ( !len( rbFullPath ) ) {
+			throw(
+				"The resource bundle file: #rbFilePath# does not exist. Please check your path",
+				"FullPath: #rbFullPath#",
+				"ResourceBundle.InvalidBundlePath"
+			);
+		}
+
+		// read file
+		var fis = createObject( "java", "java.io.FileInputStream" ).init( rbFullPath );
```

Comment:
This code is the same as in the method above, I think we need to create a private method to encapsulate the loading of the resource file. so both methods can reuse it.

Reply:
method getResourceFileInputStream created. Used it in 3 different methods